extractor/src/main/java/org/schabi/newpipe/extractor/services/bandcamp/extractors/BandcampChannelExtractor.java

```diff
@@ -107,29 +115,21 @@ public boolean isVerified() throws ParsingException {
 
     @Nonnull
     @Override
-    public InfoItemsPage<StreamInfoItem> getInitialPage() throws ParsingException {
-
-        final StreamInfoItemsCollector collector = new StreamInfoItemsCollector(getServiceId());
-
+    public List<ListLinkHandler> getTabs() throws ParsingException {
         final JsonArray discography = channelInfo.getArray("discography");
 
-        for (int i = 0; i < discography.size(); i++) {
-            // A discograph is as an item appears in a discography
-            final JsonObject discograph = discography.getObject(i);
+        final List<ListLinkHandler> tabs = new ArrayList<>();
+        tabs.add(new ReadyChannelTabListLinkHandler(getUrl(), getId(),
+                ChannelTabs.TRACKS, this::buildTracksTabExtractor));
 
-            if (!discograph.getString("item_type").equals("track")) {
-                continue;
-            }
-
-            collector.commit(new BandcampDiscographStreamInfoItemExtractor(discograph, getUrl()));
+        if (discography.stream().anyMatch(o -> (
+                (JsonObject) o).getString("item_type").equals("album"))) {
+            tabs.add(new BandcampChannelTabHandler(
+                    getUrl() + BandcampChannelTabLinkHandlerFactory.URL_SUFFIX,
+                    getId(), ChannelTabs.ALBUMS, discography));
```

Comment:
Yes, we definetly should display albums and single tracks on the same page! Unlike with YouTube this causes no duplicates because playlists can only contain tracks that are not also displayed as a single track. (Except if an artist publishes the same track twice, which is besides the point.)

Reply:
I would not merge the two tabs. It would not fit into our tab type scheme (we have album and track tabs, how should we call the merged one?).
There would also be issues with the playlist controls to play all tracks of a channel.

Reply:
No, don't worry, the client part is simple to fix (a `.filter(StreamInfoItem.class::isinstance)` on the returned items is enough). So I would keep one tab.

Reply:
Actually, I don't know. What is @AudricV's opinion?

Reply:
I think if split it up, it would be more effort for a user to explore an artist's creations, because – unlike YouTube – there is no difference in how it is used between a track and an album; on YouTube you look at the Videos tab to see the latest videos and at the Playlists tab to specifically look for videos on a certain topic. Whereas on Bandcamp it doesn't matter whether your next favorite song is part of an album or not.

There's no reason we should make a distinction where the website doesn't make one.

> we have album and track tabs, how should we call the merged one?

Discography is the term we're looking for.

Reply:
With Bandcamp, we are not fetching the desktop page but are using the mobile API which returns a list of discographs (both tracks and albums). The album items retuned by the API do not contain a track list, so it would not be possible (without further requests) do display all tracks of an artist in a single list.

We could merge the items together and have a list of both tracks and albums. But I dont think that would be very intuitive. Clicking "Play all" for example would only play the tracks and ignore the albums. If an artist does not have single tracks, the feature would not work at all and the user might get confused, since there are items present in the list.

Reply:
That sounds like a matter of adapting Play All to work with playlists in my eyes, if that is the only unintuitive thing you see here. This would result in the behavior that a user expects in all regards (i.e. tracks and albums on one page like on the website, as well as being able to play all of an artist).

Reply:
I agree it would be off-topic to make the Play All button behave properly, so ok, let's keep them two separate tabs (but they should be made so that the channel page is fetched only once).

Reply:
This is already the case, bandcamp channels retun up to 2 ReadyChannelTabListLinkHandlers